etcdutl/snapshot/v3_snapshot.go

```diff
@@ -138,31 +140,45 @@ func (s *v3Manager) Status(dbPath string) (ds Status, err error) {
 		if v != nil {
 			ds.Version = v.String()
 		}
+		var kv mvccpb.KeyValue
 		c := tx.Cursor()
 		for next, _ := c.First(); next != nil; next, _ = c.Next() {
 			b := tx.Bucket(next)
 			if b == nil {
-				return fmt.Errorf("cannot get hash of bucket %s", string(next))
+				return fmt.Errorf("nil bucket: %q", string(next))
 			}
-			if _, err = h.Write(next); err != nil {
-				return fmt.Errorf("cannot write bucket %s : %v", string(next), err)
+			_, err = h.Write(next)
+			if err != nil {
+				return fmt.Errorf("cannot hash bucket: name=%q err=%w", string(next), err)
 			}
-			iskeyb := (string(next) == "key")
+
+			iskeyb := (bytes.Equal(next, schema.Key.Name()))
 			if err = b.ForEach(func(k, v []byte) error {
-				if _, herr := h.Write(k); herr != nil {
-					return fmt.Errorf("cannot write to bucket %s", herr.Error())
+				_, err = h.Write(k)
+				if err != nil {
+					return fmt.Errorf("cannot hash bucket: key=%q err=%w", k, err)
 				}
-				if _, herr := h.Write(v); herr != nil {
-					return fmt.Errorf("cannot write to bucket %s", herr.Error())
+				_, err = h.Write(v)
+				if err != nil {
+					return fmt.Errorf("cannot hash bucket: key=%q value=%q err=%w", k, v, err)
 				}
 				if iskeyb {
-					rev := mvcc.BytesToRev(k)
+					var rev mvcc.Revision
+					rev, err = bytesToRev(k)
+					if err != nil {
+						return fmt.Errorf("cannot parse revision key: key=%q err=%w", k, err)
+					}
 					ds.Revision = rev.Main
+
+					err = kv.Unmarshal(v)
```

Comment:
would you mind put `var kv mvccpb.KeyValue` here? Just thinking that it can be easy to review.

Reply:
Done.